lib/tools/adb-commands.js

```diff
@@ -0,0 +1,34 @@
+let methods = {};
+
+methods.getAdbWithCorrectAdbPath = async function () {
+  try {
+    this.adb.path = await this.sysCallsHelper.getSdkBinaryPath("adb");
+    this.binaries.adb = this.adb.path;
+  } catch (err) {
+    return err;
```

Comment:
this is a very strange pattern. should probably catch inside whoever's calling this method instead